src/wallet.py

```diff
@@ -0,0 +1,92 @@
+import json
+from json.decoder import JSONDecodeError
+
+import structlog
+from nacl.encoding import HexEncoder
+from nacl.signing import SigningKey
+import base64
+from cryptography.fernet import Fernet
+
+
+logger = structlog.get_logger(__name__)
+
+class wallet:
+    PRIVATE_KEY = ""
```

Comment:
Perhaps we are missing the constructor.
```python
def __init__(self): 
    self.privatekey = ""
    self.publickey = ""
```

Maybe we can store only the public key and the encripted key. So every time the user wants to do something with the wallet he/she needs to provide the password.